### PR TITLE
Update to OpenSSL 1.1.1l

### LINKED ISSUE #328: Update to OpenSSL 1.1.1l when released
Expected August 25, 2021

### PR DESCRIPTION
**Pull request by dstebila**
Fixes #328

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -13,21 +13,19 @@ The **Open Quantum Safe (OQS) project** has the goal of developing and prototypi
...
 =============
 
-This is the 2021-08 snapshot release of OQS-OpenSSL, which was released on August 11, 2021. This release is intended to be used with liboqs version 0.7.0.
+This is development branch for the 2021-xx snapshot release of OQS-OpenSSL. This release is intended to be used with liboqs version 0.7.0.
 
 What's New
 ----------
 
-This is the sixth snapshot release of OQS-OpenSSL_1\_1\_1-stable.  It is based on OpenSSL 1.1.1k.
+This is the seventh snapshot release of OQS-OpenSSL_1\_1\_1-stable.  It is based on OpenSSL 1.1.1l.
```
Comment (baentsch) [inline RELEASE.md:21]: What about dropping this statement in favour of logging the OSSL release ID below?

File: RELEASE.md [hunk 2]
```diff
@@ -13,21 +13,19 @@ The **Open Quantum Safe (OQS) project** has the goal of developing and prototypi
...
-- Updates algorithms to those used in liboqs 0.7.0, as described in the [liboqs release notes](https://github.com/open-quantum-safe/liboqs/blob/main/RELEASE.md).
-- Improves signing support with different digest algorithms
-- Enables building OQS-OpenSSL for UEFI/EDKII (contributed by Jiewen Yao, Intel)
-- Improved documentation of algorithm identifiers
+- Update OpenSSL to version 1.1.1l.
 
 Previous release notes
 ----------------------
 
+- [OQS-OpenSSL 1.1.1 snapshot 2021-08](https://github.com/open-quantum-safe/openssl/releases/tag/OQS-OpenSSL_1_1_1-stable-snapshot-2021-08) aligned with liboqs 0.7.0 (August 11, 2021)
```
Comment (baentsch) [inline RELEASE.md]: (also) reference OSSL release ID (only) here.

**Review by baentsch — APPROVED** [review]
Basically LGTM. Minor comment suggestions added.

File: RELEASE.md (hunk 2 above)
Comment (dstebila) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 1 above)
Comment (dstebila) [reply RELEASE.md:21]: Keeping it here since it only goes in the list below after the next release comes out.

**Pull request merged**